pkg/kapp/diff/versioned_resource.go

```diff
@@ -34,6 +40,12 @@ func (d VersionedResource) BaseNameAndVersion() (string, string) {
 	if len(pieces) > 1 {
 		return strings.Join(pieces[0:len(pieces)-1], nameSuffixSep), pieces[len(pieces)-1]
 	}
+	if d.StripNameHashSuffix() {
```

Comment:
Yes, I intend to add a (customizeable) filter for CMs and Secrets. :)

My intention is to "hide" stripping the suffix behind a flag (in Config YAML), targeted at kustomize users. I inted that the customizeable filter above allows exluding resources by resourceMatchers. Hiding it behind a flag means users will need to opt-in, where they then should be aware of the potential side effects and how to fix the unwanted ones via config. That would be my take on the latter. But the code and demo (e2e) tests I intend to write should be better at explaining that :)

Reply:
@praveenrewar While it is true that kustomize only creates hash-suffixes for ConfigMaps and Secrets when only using the [built-in plugins](https://github.com/kubernetes-sigs/kustomize/tree/master/plugin), kustomize can be [extended with custom plugins](https://kubectl.docs.kubernetes.io/guides/extending_kustomize/) that may also use the built-in [hash-suffix-transformer](https://github.com/kubernetes-sigs/kustomize/blob/master/plugin/builtin/hashtransformer/HashTransformer.go).

In practice I've never seen someone actually doing that, but I wanted to add this information for the sake of completeness.

Reply:
> Yes, I intend to add a (customizeable) filter for CMs and Secrets. :)

Thanks you :)

> Hiding it behind a flag means users will need to opt-in, where they then should be aware of the potential side effects and how to fix the unwanted ones via config

True. Just mentioned thinking that it would be great to have less side effects to begin with :)

Reply:
> While it is true that kustomize only creates hash-suffixes for ConfigMaps and Secrets when only using the [built-in plugins](https://github.com/kubernetes-sigs/kustomize/tree/master/plugin), kustomize can be [extended with custom plugins](https://kubectl.docs.kubernetes.io/guides/extending_kustomize/) that may also use the built-in [hash-suffix-transformer](https://github.com/kubernetes-sigs/kustomize/blob/master/plugin/builtin/hashtransformer/HashTransformer.go).

Ah, I see. So I guess we won't be able to have that check then 😞